.github/workflows/terraform_gh_runner.yml

```diff
@@ -19,40 +21,47 @@ jobs:
       - name: Set up Terraform
         uses: hashicorp/setup-terraform@v1
         with:
-          terraform_version: "1.4.2"
+          terraform_version: "1.5.7"
           cli_config_credentials_token: ${{ secrets.TF_CLOUD_TOKEN }}
 
       - name: Install dependencies
         run: |
           # Install any dependencies required by your Terraform code
 
       - name: Run Bash Script
+        env:
+          GH_TOKEN: ${{ github.token }}
         run: |
-            # GitHub repository and access token
-            repo="subspace/infra"
-            token=${{ secrets.PAT_TOKEN }}
+          # GitHub repository and access token
+          repo="subspace/infra"
+          token=${{ secrets.PAT_TOKEN }}
 
-            # API endpoint
-            url="https://api.github.com/repos/$repo/actions/runners/registration-token"
+          # API endpoint
+          url="https://api.github.com/repos/$repo/actions/runners/registration-token"
 
-            # Send POST request to get the registration token
-            response=$(curl -X POST -H "Authorization: token $token" -s "$url")
+          # Send POST request to get the registration token
+          response=$(curl -X POST -H "Authorization: token $token" -s "$url")
 
-            # Extract the token value from the response
-            runner_token=$(echo "$response" | jq -r '.token')
+          # Extract the token value from the response
+          runner_token=$(echo "$response" | jq -r '.token')
 
-            # Export the token as an environment variable
-            echo "export RUNNER_TOKEN=$runner_token" >> $GITHUB_ENV
+          # Export the token as an environment variable
+          echo "export RUNNER_TOKEN=$runner_token" >> $GITHUB_ENV
 
-            # Set the runner token as an environment variable
-            export RUNNER_TOKEN="$runner_token"
+          # Set the runner token as an environment variable
+          export RUNNER_TOKEN="$runner_token"
 
-            # Store the token as a secret in GitHub Actions
-            gh secret set RUNNER_TOKEN -r "$repo" -b "$runner_token"
+          # Store the token as a secret in GitHub Actions
+          gh secret set RUNNER_TOKEN -r "$repo" -b "$runner_token"
```

Comment:
This makes little sense, why are we setting it as environment variable in two different ways AND secrets? We only seem to use the first environment variable in the next step.

All this token management looks like a complete mess to me. A bunch of random things, some of which are used and some are not, but not clear why.

Comments were supposed to help, but these are exactly the kinds of comments that shouldn't be written in the first place: they say exactly the same thing that code does, so I essentially read the same information twice, which isn't helpful. Comments should add useful information, explaining why things are done, how they are useful but none of this is currently present.

Reply:
There are essentially 3 tokens, secrets.GITHUB_TOKEN for using github api. `PAT_TOKEN` to access repository and create a runner registration token.

Reply:
But why do you store it in the environment variable at all? It is already stored in secrets. Or why is it stored in secrets if it is only necessary in environment variable for the next step. Also storing it in enviromnent variable probably means it will be printed in CI run in logs, which is a security issue.

This is why it looks like a mess.

Reply:
Storing as the env variable in that step will not print it to CI, however I have removed it and just used the one from the secret at https://github.com/subspace/infra/pull/183/files/99a5ac25712fdda531fd88026a12f96aafeacb83#diff-b46b68b6df852ad5f8fc96162f55c2fe198f6d53eb2fe7d7ce12fcd4b2650ba6R67

Reply:
I am asking about `$runner_token` that this workflow stores in secrets, not PAT.

Reply:
The token expires after one hour.
https://docs.github.com/en/rest/actions/self-hosted-runners?apiVersion=2022-11-28#create-a-registration-token-for-an-organization